src/OpenTelemetry.Extensions.PersistentStorage.Abstractions/Storage.cs

```diff
@@ -0,0 +1,71 @@
+﻿// <copyright file="Storage.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Extensions.PersistentStorage.Abstractions
+{
+    /// <summary>
+    /// Represents persistent storage.
+    /// </summary>
+    public abstract class Storage
+    {
+        /// <summary>
+        /// Reads a sequence of blobs from storage.
+        /// </summary>
+        /// <returns>
+        /// List of blobs if present in storage or else null.
```

Comment:
Might be nicer for us to return empty IEnumerable instead of null if nothing is found.